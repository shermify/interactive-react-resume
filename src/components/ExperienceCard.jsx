import React from 'react';
import PropTypes from 'prop-types';

const ExperienceCard = (props) => {
  const { company, date, description } = props;
  return (
    <div className="card" key={company}>
      <div className="card-block">
        <h4 className="card-title">{company}</h4>
        <h6 className="card-subtitle mb-2 text-muted">{date}</h6>
        <div className="card-text text-justify">
          {description.paragraphs.map((paragraph, i) =>
            (<p key={i}>{paragraph}</p>)) // eslint-disable-line react/no-array-index-key
          }
        </div>
      </div>
    </div>
  );
};

ExperienceCard.propTypes = {
  company: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  description: PropTypes.shape({
    paragraphs: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
};

export default ExperienceCard;
