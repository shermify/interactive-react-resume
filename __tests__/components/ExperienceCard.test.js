import React from 'react';
import renderer from 'react-test-renderer';
import ExperienceCard from '../../src/components/ExperienceCard';

describe('ExperienceCard', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ExperienceCard
        company="test company"
        date="1/1/2017"
        description={{ paragraphs: ['description 1', 'description 2'] }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
