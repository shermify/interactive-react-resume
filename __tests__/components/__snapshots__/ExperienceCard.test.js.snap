// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceCard renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-block"
  >
    <h4
      className="card-title"
    >
      test company
    </h4>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
      1/1/2017
    </h6>
    <div
      className="card-text text-justify"
    >
      <p>
        description 1
      </p>
      <p>
        description 2
      </p>
    </div>
  </div>
</div>
`;
